test/e2e/common/node/downwardapi.go

```diff
@@ -414,6 +416,156 @@ var _ = SIGDescribe("Downward API", framework.WithSerial(), framework.WithDisrup
+				},
+			}
+			testDownwardAPIUsingPod(ctx, f, pod, env, expectations)
+		})
+
+		ginkgo.It("should provide default limits.cpu/memory from pod level resources or node allocatable", func(ctx context.Context) {
+			podName := "downward-api-" + string(uuid.NewUUID())
+			cName := "dapi-container"
+			env := []v1.EnvVar{
+				{
+					Name: "CPU_LIMIT",
+					ValueFrom: &v1.EnvVarSource{
+						ResourceFieldRef: &v1.ResourceFieldSelector{
+							Resource: "limits.cpu",
+						},
+					},
+				},
+				{
+					Name: "MEMORY_LIMIT",
+					ValueFrom: &v1.EnvVarSource{
+						ResourceFieldRef: &v1.ResourceFieldSelector{
+							Resource: "limits.memory",
+						},
+					},
+				},
+			}
+
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   podName,
+					Labels: map[string]string{"name": podName},
+				},
+				Spec: v1.PodSpec{
+					Resources: &v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceCPU: resource.MustParse("1250m"),
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:    cName,
+							Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+							Command: []string{"sh", "-c", "env"},
+							Env:     env,
+						},
+					},
+					RestartPolicy: v1.RestartPolicyNever,
+				},
+			}
+
+			// Before the Pod Level Resources feature was introduced, if container-level
+			// resource limits were not set, the Downward API would return the node's
+			// allocatable resources instead of 0. So, it was sufficient to simply check
+			// that the value set in the environment variable was non-zero.
+			// However, after the Pod Level Resources feature was introduced, the node's
+			// allocatable resources are used as a fallback when pod-level resource limits
+			// are not specified. Therefore, we now need to verify that the value
+			// in the environment variable matches the node's allocatable resources.
+			// Because we need to dynamically retrieve the node's allocatable resources,
+			// we cannot simply use e2epodoutput.TestContainerOutputRegexp here.
+			podClient := e2epod.NewPodClient(f)
+			createdPod := podClient.Create(ctx, pod)
+			ginkgo.DeferCleanup(func(ctx context.Context) {
+				ginkgo.By("delete the pods")
+				podClient.DeleteSync(ctx, createdPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
+			})
+
+			// Wait for client pod to complete.
+			err := e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, f.ClientSet, createdPod.Name, f.Namespace.Name, f.Timeouts.PodStart)
+			framework.ExpectNoError(err)
+
+			// Grab its logs.  Get host first.
+			podStatus, err := podClient.Get(ctx, createdPod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// Get the node allocatable resources
+			nodeName := podStatus.Spec.NodeName
+			node, err := f.ClientSet.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			// Prepare expectations based on node allocatable resources
+			expectations := []string{
+				"CPU_LIMIT=2",
```

Comment:
nit - add a comment about why 1250m becomes 2

Reply:
I actually just copied another test case, but now that I understand the reason. Added a comment explaining it.